rfcs/prevent-ci-trigger-on-doc-prs.md

```diff
@@ -0,0 +1,64 @@
+# RFC: Prevent CI and code related tasks from triggering on doc/rfc PRs
+
+---
+
+_Contributors: @andrefcdias_
+
+## Summary
+
+The purpose of this RFC is to find a solution to prevent unnecessary GitHub checks from running on PRs that only affect documentation.
+
+## Problem statement
+
+Whenever someone creates a PR, independently of the content, the CI is triggered and added as a check for a successfull PR. This, along with Screener and CodeSandbox, is consuming unnecessary resources on our pipeline for, essentially, no reason.
+
+Apart from this, other PRs that actually need CI checks, will be delayed due to the lack of parallelism in our pipeline.
+By avoiding these checks, we can prevent a wasteful CI pipeline and accelerate work in PRs that bring value to our users.
+
+## Detailed Design or Proposal
+
+### Azure DevOps (Build, Deploy, SizeAuditor, Perf Analysis, BundleSize)
+
+To prevent the builds, adding the following will prevent unnecessary triggers:
+
+```yaml
+pr:
+  branches:
+    include:
+      - master
+  paths:
+    exclude:
+      - rfcs
```

Comment:
**thought:** this is great starting point. In future I'd like to have proper tooling in place (content aware, so these kinds of "imperative" rules would be mitigated - for example - nx dev tools handles this very well)

Reply:
From the docs, there's no other way with the `pr` trigger to manage this. Would we then use nx dev tools on our GitHub checks to validate/build/test trigger the build/deploy on merge completion for example?

Reply:
> From the docs, there's no other way with the pr trigger to manage this

no worries I get that. thx!

> Would we then use nx dev tools on our GitHub checks to validate/build/test and then trigger the build/deploy on merge completion for example?

yes. 

**"short" example (out off topic):**
- You'd run something like `nx affected --target=build`
- This would execute `build` only on affected projects in tree. 
- In our case , lets say we would have 3 projects (rfcs, spec, docs) which are not dependent on each other. 
- Our affected command would run only `build` for `rfcs` project.
-  if such a  target exists for particular project - `build` (in this case it doesn't), `affected` run would finish in matter of miliseconds :) 🚀

Reply:
I know nx is powerful but I don’t see us switching to it any time soon. And even if we did it’s not clear from what you said whether it would solve the issue with required checks. (If it does provide a way to still satisfy the required checks, maybe we could look at what it does and apply a similar solution.)

Also in case you weren’t aware, we do have some level of content-aware builds for the sub-steps that are orchestrated with lage, but that doesn’t help with everything.

Reply:
Nx solution would mitigate the issue like the solution you mentioned below. We would have the nx check and it would pass immediately since `docs` wouldn't have a build task.
However, unless we there's a way to integrate the screener/codesandbox checks in nx, this would still be there.

Reply:
> unless we there's a way to integrate the screener/codesandbox checks in nx

that's quite easy to do 💪